src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/ClientCertificateStore.java

```diff
@@ -0,0 +1,177 @@
+
+        if (passphrase == null) {
+            passphrase = CertificateHelper.generateRandomPassphrase();
+            runtimeConfiguration.updateClientPassphrase(passphrase);
+        }
+
+        this.passphrase = passphrase.toCharArray();
+
+        try {
+            this.keyStore = loadStore();
+        } catch (KeyStoreException | IOException | CertificateException | NoSuchAlgorithmException
+                 | UnrecoverableKeyException e) {
+            this.keyStore = createStore();
+        }
+    }
+
+    private KeyStore loadStore() throws KeyStoreException, IOException,
+            CertificateException, NoSuchAlgorithmException, UnrecoverableKeyException {
+        KeyStore ks = KeyStore.getInstance("JKS");
+        try (InputStream ksInputStream = Files.newInputStream(storePath)) {
+            ks.load(ksInputStream, passphrase);
+        }
+
+        return ks;
+    }
+
+    private KeyStore createStore() throws KeyStoreException {
+        KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
+
+        try {
+            ks.load(null, null);
+            flushToDisk(ks);
+        } catch (IOException | NoSuchAlgorithmException | CertificateException e) {
+            throw new KeyStoreException("unable to load client keystore", e);
+        }
+
+        return ks;
+    }
+
+    /**
+     * Writes the KeyStore into the file system.
+     * TODO: Do this async we don't have to block.
+     */
+    private void flushToDisk(KeyStore ks) throws KeyStoreException {
+        try {
+            Files.createDirectories(storePath.getParent());
+
+            try (OutputStream writeStream = Files.newOutputStream(storePath)) {
+                ks.store(writeStream, passphrase);
+            }
+
+            Platform platform = Platform.getInstance();
+            FileSystemPermission accessLevel = FileSystemPermission.builder()
+                    .ownerRead(true).ownerWrite(true).build();
+
+            platform.setPermissions(accessLevel, storePath);
+        } catch (CertificateException | IOException | NoSuchAlgorithmException e) {
+            throw new KeyStoreException(e);
+        }
+    }
+
+
+    /**
+     * Stores the PEM for a certificate.
+     * @param certificateId - A Certificate ID
+     * @param certificatePem - The Pem string of the certificate
+     *
+     * @throws CertificateException - If fails generate certificate from PEM
+     * @throws KeyStoreException - If fails to store the key store into disk
+     */
+    public void storePem(String certificateId, String certificatePem) throws CertificateException, KeyStoreException {
+        CertificateFactory cf = CertificateFactory.getInstance("X.509");
+        InputStream targetStream = new ByteArrayInputStream(certificatePem.getBytes());
+        Certificate cert = cf.generateCertificate(targetStream);
+        keyStore.setCertificateEntry(certificateId, cert);
+        flushToDisk(keyStore);
```

Comment:
Not super happy about this given we are writing the entire thing to disk synchronously. I will improve and do the flushing async on a next PR.

Reply:
Treat this as a nice to have. Not needed for now, IMO.

Reply:
Yeah the downside to not having it would be that we are tracking a certificate and we can't find it's PEM stored after a reboot and we would have to wait for the client to try to get verified again so we register the cert.

I will leave it for now I can always remove it if it is a problem